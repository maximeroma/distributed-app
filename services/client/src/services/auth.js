import axios from "axios"

const USERS_SERVICE_URL = process.env.REACT_APP_USERS_SERVICE_URL

export const signUp = data =>
  axios
    .post(`${USERS_SERVICE_URL}/auth/register`, data)
    .then(({data}) => data)
    .catch(err => console.log(err))

export const login = data =>
  axios
    .post(`${USERS_SERVICE_URL}/auth/login`, data)
    .then(({data}) => data)
    .catch(err => console.log(err))
