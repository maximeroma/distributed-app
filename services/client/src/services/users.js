import axios from "axios";

const url = process.env.REACT_APP_USERS_SERVICE_URL;

export const getUsers = () => axios.get(`${url}/users`);
