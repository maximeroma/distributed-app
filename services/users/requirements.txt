Click==7.0
Flask==1.0.2
Flask-SQLAlchemy==2.3.2
itsdangerous==1.1.0
Jinja2==2.11.3
MarkupSafe==1.1.0
psycopg2-binary==2.7.7
SQLAlchemy==1.3.0
Werkzeug==0.14.1
Flask-Testing==0.6.2
gunicorn==19.9
coverage==4.5.2
flake8==3.6.0
flask-debugtoolbar==0.10.1
flask-cors==3.0.7
flask-migrate==2.3.1
flask-bcrypt==0.7.1
pyjwt==1.7.1
pytest-flask==0.10.0
pytest-cov==2.6.1
graphene==2.0
flask-graphql==2.0.0
graphene-sqlalchemy==2.0
