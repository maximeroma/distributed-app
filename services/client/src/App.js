import React, {Component} from "react"
import {Router} from "@reach/router"
import Home from "routes/home"
import About from "routes/about"
import Logout from "routes/logout"
import NavBar from "components/navbar"
import Login from "routes/log-in"
import UserStatus from "routes/status"
import Register from "routes/register"
import {signUp, login} from "services/auth"
import {AuthProvider} from "hooks/auth"
import "./App.css"

export default () => {
  return (
    <div>
      <AuthProvider>
        <Router>
          <NavBar path="/">
            <Home path="/" />
            <About path="about" />
            <Register path="/register" signUp={signUp} />
            <UserStatus path="/status" />
            <Login path="/login" login={login} />
            <Logout path="/logout" />
          </NavBar>
        </Router>
      </AuthProvider>
    </div>
  )
}
