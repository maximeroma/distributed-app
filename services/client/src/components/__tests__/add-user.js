import React from "react"
import {render, fireEvent, wait} from "react-testing-library"
import App from "../../App"
import {addUser, getUsers} from "../../services/users"

jest.mock("../../services/users", () => ({
  addUser: jest.fn(data => Promise.resolve({data})),
  getUsers: jest.fn(() =>
    Promise.resolve({
      data: {
        data: {
          users: []
        }
      }
    })
  )
}))

const flushPromises = async () => new Promise(resolve => setImmediate(resolve))

test("i can see user form correctly", () => {
  const {debug, getByLabelText} = render(<App />)

  expect(getByLabelText(/username/i)).toBeInTheDocument()
  expect(getByLabelText(/email/i)).toBeInTheDocument()
})

test("i can see user form correctly", async () => {
  const {getByLabelText, getByText, debug} = render(<App />)

  fireEvent.change(getByLabelText(/username/i), {target: {value: "maxime"}})
  fireEvent.change(getByLabelText(/email/i), {
    target: {value: "roma@gmail.com"}
  })

  fireEvent.submit(getByText(/submit/i))

  await flushPromises()
  expect(addUser).toHaveBeenCalled()
  expect(addUser.mock.calls[0][0]).toMatchSnapshot()
})
