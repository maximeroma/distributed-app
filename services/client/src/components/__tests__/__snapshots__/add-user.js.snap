// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i can see user form correctly 1`] = `
Object {
  "email": "roma@gmail.com",
  "username": "maxime",
}
`;
